service/k8sclient/internal/pxc/pxc_types.go

```diff
@@ -143,7 +143,7 @@ type PerconaXtraDBClusterList struct {
 type PodSpec struct { //nolint:maligned
 	Enabled                       bool                            `json:"enabled,omitempty"`
 	Pause                         bool                            `json:"pause,omitempty"`
-	Size                          int32                           `json:"size,omitempty"`
+	Size                          *int32                          `json:"size,omitempty"`
```

Comment:
I though that `omitempty` ignores field if it has default value, doesn't it?

Reply:
IIRC, omitempty means that it won't be included in the json by the marshaler but if you want to **detect** if it was present or not, you need a pointer

Reply:
We never want to decrease size to 0. We use `pause` field for that, so we don't have to differentiate 0 and nil.